arcaflow_plugin_fio/fio_schema.py

```diff
@@ -65,31 +62,37 @@ def is_sync(self) -> bool:
         return self.value in self._sync_io_engines
 
 
-time_pattern = re.compile(r"^\d*[d,h,m,s,ms,us]*$")
+duration_pattern_string = r"[1-9][0-9]*(?:d|h|m|s|ms|us)?"
+duration_pattern = re.compile(rf"^{duration_pattern_string}$")
+duration_range_pattern = re.compile(
+    rf"^{duration_pattern_string}(?:-{duration_pattern_string})?$"
+)
 
 size_pattern_string = (
-    r"(?:0x\d+)|"
-    r"(?:\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)*)"
+    r"0x[0-9a-fA-F]+|"
+    r"\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)?"
 )
 size_pattern = re.compile(
-    rf"^{size_pattern_string}(?:,{size_pattern_string}){{0,2}}$", re.IGNORECASE
+    rf"^(?:{size_pattern_string}(?:,{size_pattern_string}){{0,2}})$",
+    re.IGNORECASE,
 )
 
-size_pattern_with_percent_string = (
-    rf"{size_pattern_string}|(?:[1-9][0-9]?%$|^100%)"
-)
+size_pattern_with_percent_string = rf"{size_pattern_string}|[1-9][0-9]?%|100%"
 size_pattern_with_percent = re.compile(
-    rf"^{size_pattern_with_percent_string}$", re.IGNORECASE
+    rf"^(?:{size_pattern_with_percent_string})$", re.IGNORECASE
 )
```

Comment:
I'm reconsidering my earlier stance:  perhaps the grouping _does_ belong inside the string, at least when the string contains an alteration.  😒

Reply:
This one tests out OK as it currently is.